apps/zui/src/models/lake.ts

```diff
@@ -0,0 +1,42 @@
+import {DomainModel} from "src/core/domain-model"
+import {LakeAttrs} from "../js/state/Lakes/types"
+
+export class Lake extends DomainModel<LakeAttrs> {
+  getAddress(): string {
+    return this.attrs.port
+      ? [this.attrs.host, this.attrs.port].join(":")
+      : this.attrs.host
+  }
+
+  serialize(): LakeAttrs {
+    return {...this.attrs}
+  }
+
+  get authType() {
+    return this.attrs.authType
+  }
+
+  get authData() {
+    return this.attrs.authData
+  }
+
+  get name() {
+    return this.attrs.name
+  }
+
+  get id() {
+    return this.attrs.id
+  }
+
+  get host() {
+    return this.attrs.host
+  }
+
+  get port() {
+    return this.attrs.port
+  }
+
+  get version() {
+    return this.attrs.version
+  }
+}
```

Comment:
This is the I file I changed and moved. Everything else is just updating references and slight usage.